### PR TITLE
Fix SortButton labels.

### LINKED ISSUE #5154: SortButton translate reference
**What you were expecting:**
The SortButton should have labels based on the resource, instead the code of SortButton uses the "products" resources, code should be fixed, otherwise cannot be used.

**Related code:**
Line 74 ```field: translate(`resources.products.fields.${currentSort.field}`)```
Line 117 ```{translate(`resources.products.fields.${field}`)}```

**Environment**

* React-admin version: 3.8.1
* Last version that did not exhibit the issue (if applicable): /
* React version: 
* Browser:
* Stack trace (in case of a JS error):

### PR DESCRIPTION
**Pull request by WiXSL**
Fixes #5154

### CODE REVIEW

File: packages/ra-ui-materialui/src/button/SortButton.tsx [hunk 1]
```diff
@@ -71,8 +71,8 @@ const SortButton: FC<{ fields: string[]; label?: string }> = ({
     };
 
     const buttonLabel = translate(label, {
-        field: translate(`resources.products.fields.${currentSort.field}`),
-        order: translate(`ra.sort.${currentSort.order}`),
+        field: currentSort.field,
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/button/SortButton.tsx:74]: no, we must grab the `resource` from the listcontext, and replace `products` with it

Comment (WiXSL) [reply packages/ra-ui-materialui/src/button/SortButton.tsx]: Right

File: packages/ra-ui-materialui/src/button/SortButton.tsx [hunk 2]
```diff
@@ -71,8 +72,8 @@ const SortButton: FC<{ fields: string[]; label?: string }> = ({
     };
 
     const buttonLabel = translate(label, {
-        field: translate(`resources.products.fields.${currentSort.field}`),
-        order: translate(`ra.sort.${currentSort.order}`),
+        field: translate(`resources.${resource}.fields.${currentSort.field}`),
+        order: translate(`resources.${resource}.fields.${currentSort.order}}`),
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/button/SortButton.tsx:76]: no, this one shouldn't be changed

File: packages/ra-ui-materialui/src/button/SortButton.tsx [hunk 3]
```diff
@@ -114,7 +115,7 @@ const SortButton: FC<{ fields: string[]; label?: string }> = ({
                         data-sort={field}
                         key={field}
                     >
-                        {translate(`resources.products.fields.${field}`)}{' '}
+                        {translate(field, { _: field })}{' '}
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/button/SortButton.tsx:118]: please do just like on line 75

File: packages/ra-ui-materialui/src/button/SortButton.tsx [hunk 4]
```diff
@@ -43,6 +43,7 @@ const SortButton: FC<{ fields: string[]; label?: string }> = ({
     fields,
     label = 'ra.sort.sort_by',
 }) => {
+    const { resource } = useListContext();
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/button/SortButton.tsx:46]: no need to call 2 contexts, please move currentSort and setSort here and remove useListSortContext

File: packages/ra-ui-materialui/src/button/SortButton.tsx (hunk 2 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/button/SortButton.tsx]: Done

File: packages/ra-ui-materialui/src/button/SortButton.tsx (hunk 3 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/button/SortButton.tsx]: Done

File: packages/ra-ui-materialui/src/button/SortButton.tsx (hunk 4 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/button/SortButton.tsx]: Done

File: packages/ra-ui-materialui/src/button/SortButton.tsx [hunk 5]
```diff
@@ -114,7 +114,7 @@ const SortButton: FC<{ fields: string[]; label?: string }> = ({
                         data-sort={field}
                         key={field}
                     >
-                        {translate(`resources.products.fields.${field}`)}{' '}
+                        {translate(field)}{' '}
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/button/SortButton.tsx:117]: no, this should be `resources.${resource}.fields.${field}`

Comment (WiXSL) [reply packages/ra-ui-materialui/src/button/SortButton.tsx]: Done

File: packages/ra-ui-materialui/src/button/SortButton.tsx [hunk 6]
```diff
@@ -114,7 +114,7 @@ const SortButton: FC<{ fields: string[]; label?: string }> = ({
                         data-sort={field}
                         key={field}
                     >
-                        {translate(field)}{' '}
+                        {translate(`resources.${resource}.fields.${field}`}{' '}
```
Comment (Luwangel) [inline packages/ra-ui-materialui/src/button/SortButton.tsx:117]:
```suggestion
                        {translate(`resources.${resource}.fields.${field}`)}{' '}
```

The translate function is not closed ;)

Comment (WiXSL) [reply packages/ra-ui-materialui/src/button/SortButton.tsx]: I'm really sorry. Thank you for your patience.

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**